lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -265,6 +276,63 @@ public void run() {
 		}
 	}
 
+	/**
+	 * Authentication token changes for the current connection are possible when
+	 * the client is in the {@link ConnectionState#connected}, {@link ConnectionState#connecting}
+	 * or {@link ConnectionState#disconnected } states.
+	 *
+	 * In the current protocol version we are not able to update auth params on the fly;
+	 * so disconnect, and the new auth params will be used for subsequent reconnection
+	 * * <p>
+	 * Spec: RTC8 (reauthorise)
+	 * </p>
+	 *
+	 */
+	public void onAuthUpdated() {
+		switch (state.state){
+			case connected:
+			case connecting:
+			case disconnected:
+				connection.once(ConnectionState.failed, new ConnectionStateListener() {
+					@Override
+					public void onConnectionStateChanged(ConnectionStateChange state) {
+						synchronized (ConnectionManager.this) {
+							ConnectionManager.this.notify();
+						}
+
+					}
+				});
+				disconnect();
```

Comment:
Instead of this there should be an explicit call to close the transport (without closing the connection), which will cause the transport to disconnect and be disposed. It should not be initiated by changing connection manager state.